### DISCUSSION TITLE
Request: onClick handler for BarList row items

### DISCUSSION THREAD

**Discussion by jschuur** [tremorlabs/tremor-npm#803] [Ideas]
I'd love to be able to specify an onClick handler that's attached to each `BarList` row item. I see that `href` was added, but that takes in a string and creates links with a `_blank` target.

I'm thinking something like this:

```jsx
<BarList data={data} onClick={(item) => console.log(`you clicked on ${item.name}`)} />
```

In this case, the`item` param would give me access to the whole element from the original `data` array, including any other non standard fields, so I can access meta data that isn’t actually displayed in the UI.

My use case right now is that I want to update a state stored in context when clicking on a BarList item. This would then then change part of the current dashboard view.

**Comment by devBenWolf** [comment]
Was thinking of something like this for tracker sections.
